src/buildstream/sourcemirror.py

```diff
@@ -93,6 +93,13 @@ class SourceMirror(Plugin):
     # The SourceMirror plugin type is only supported since BuildStream 2.2
     BST_MIN_VERSION = "2.2"
 
+    COMMON_CONFIG_KEYS = ["name", "kind"]
```

Comment:
I've always considered this to be an imperfection which we did exclusively for `Source` definitions, and feel regret for this inconsistency, it seems delicate to try to share the namespace between the core object and subclasses.

I would be more comfortable with:

```yaml
name: sonya
kind: pony
config:
   fur: pink
   tail: yellow
```

Reply:
The main reason I did this is to be able to implement the "legacy" plugin as a regular plugin (so that it can have `aliases` in the main dictionary). I have since added some special casing to it, but I'm still not sure how this could be done.

Reply:
I would say:

* Don't expose `COMMON_CONFIG_KEYS` as public API
* Keep `"aliases"` as a special/different configuration for the default SourceMirror type
* Don't allow other/user implemented `SourceMirror` plugins to extend configuration in the main `mirror` namespace (only under `config`)

I'd be happy if the base class simply made a special case for the `"default"` SourceMirror plugin type, passing it the `node` directly to `self._configure()` rather than passing the `config` dictionary for every other case.